modules/core/src/main/java/org/apache/ignite/spi/deployment/local/LocalDeploymentSpi.java

```diff
@@ -65,7 +65,7 @@
  * @see org.apache.ignite.spi.deployment.DeploymentSpi
  */
 @IgniteSpiMultipleInstancesSupport(true)
-@IgniteSpiConsistencyChecked(optional = false)
+@IgniteSpiConsistencyChecked(optional = false, checkClient = false)
```

Comment:
@liyuj , I suggest a different fix here: removing this annotation completely from **LocalDeploymentSpi** class.

The root cause of the issue you've reported is that **LocalDeploymentSpi** is set by default if no other DeploymentSpi is specified.

But **LocalDeploymentSpi** according to its name is local and doesn't interfere with any other DeploymentSpi configured anywhere in the cluster. So the right fix for the issue would be to remove this annotation (**IgniteSpiConsistencyChecked**) from **LocalDeploymentSpi** completely and revert any changes on **UriDeploymentSpi**.

Also it would be great if you turn your reproducer into a regular Ignite test. A good place for the test is **ignite-urideploy** module where **UriDeploymentSpi** is available.